src/databases/unv/avtunvFileFormat.C

```diff
@@ -5492,6 +5510,186 @@ avtunvFileFormat::ReadFile()
 #endif
         }
 #endif
+        else if (strcasestr(filename.c_str(), ".stl") != NULL)
+        {
+            // Standard Triangle Language or Standard Tessellation Language or stereolithography
+            handle = fopen(filename.c_str(), "r");
+            if (handle == NULL)
+            {
+                EXCEPTION1(InvalidDBTypeException, "This mesh file could not be openend.");
+            }
+#if INTERACTIVEREAD
```

Comment:
This two-level nesting of a compile-time `#if` and a run-time `if` seems unnecessary and IMHO detracts from readability a bit.

It seems like sometimes you want to see stuff on `stdout` instead of VisIt's normal debug logs. `debug1`, `debug2`, etc. are just CPP macro symbols defined with `#define debug1 ...` in `DebugStream.h`.

Why not do it this way...

At the top of your file have this...

```
#if INTERACTIVEREAD
#define debug1 debuglevel>=1 && std::cout
#define debug2 debuglevel>=2 && std::cout
.
.
.
#else
#include <DebugStream.h>
#endif
```

Likewise for `INTERACTIVEPLOT`.

Then you only ever have lines like `debug3 << "Found code " << code << endl;` Sometimes, it uses VisIt's normal debug logs and sometimes it uses `std::cout` (or `std:cerr` if you wish).

Reply:
Thanks for the advise, I will do that for next time. It will be much more readable !